include/grpc/support/log.h

```diff
@@ -91,7 +91,7 @@ GPRAPI void gpr_set_log_function(gpr_log_func func);
    an exception in a higher-level language, consider returning error code.  */
 #define GPR_ASSERT(x)                                 \
```

Comment:
If assert is proven to be a problem, maybe we should introduce a GPR_DEBUG_ASSERT

Reply:
Sure GPR_DEBUG_ASSERT might be something useful but GPR_LIKELY can potentially be used in many other places.

Reply:
In scenarios, where GPR_ASSERT is the right thing to do, GPR_LIKELY should help.

Reply:
I would leave out this change in GPR_ASSERT for now, and you can do a second PR toying with it to properly measure the impact.